sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -1426,4 +1426,22 @@ class CachedTableSuite extends QueryTest with SQLTestUtils
       assert(e.message.contains(s"Table or view not found: $t"))
     }
   }
+
+  test("SPARK-34347: cascading cache invalidation - SQLContext.uncacheTable") {
+    Seq(true, false).foreach { storeAnalyzed =>
+      withSQLConf(SQLConf.STORE_ANALYZED_PLAN_FOR_VIEW.key -> storeAnalyzed.toString) {
+        withTempView("view1", "view2") {
+          sql("CREATE TEMPORARY VIEW view1 AS SELECT * FROM testData WHERE key > 1")
```

Comment:
How about `GLOBAL TEMPORARY VIEW` as well.

Reply:
Good point. Will add test for that.